server/analytics.cc

```diff
@@ -36,7 +36,7 @@ namespace MLDB {
 /** Equivalent to SELECT (select) FROM (dataset) WHEN (when) WHERE (where), and each matching
     row is passed to the aggregator.
 */
-bool iterateDataset(const SelectExpression & select,
```

Comment:
This needs to continue to return whether it finished or not.  The ExpressionValueInfo should be returned as a separate element in a pair.